### PR TITLE
[EWL-4389] SG2 | Create Sidebar Articles pattern

### PR DESCRIPTION
**Pull request by chelsiejohnston**
## Ticket(s)

**Jira Ticket**
- [EWL-4389: SG2 | Create Sidebar Articles pattern](https://issues.ama-assn.org/browse/EWL-4389)

## Description
Adds a "Sidebar Articles" pattern to SG2 so that it can provide a wrapper for the corresponding custom block type in d8:

- Adds the pattern Twig file.
- Adds a `.json` file for the pattern.
- Adds a `.scss` file for the pattern.
- Adds a `.md` info file for the pattern.
- Includes the new pattern in the Topic page.
- Adds a backstop test and reference images for the pattern.

## To Test
- [ ] `gulp serve` from your local SG2 docroot
- [ ] Navigate to the [Sidebar Articles organism](http://localhost:3000/?p=organisms-sidebar-articles) pattern
  - [ ] Observe an Article Stub pattern with an image, heading, paragraph, and metadata
  - [ ] Observe an Article Stub as Small Image pattern with a small image, heading, paragraph, and metadata
  - [ ] Observe an Article Stub pattern with a heading, paragraph, and metadata only
  - [ ] Observe an Article Stub as Title Only pattern with a heading only
- [ ] Navigate to the [Topic page](http://localhost:3000/?p=pages-topic) pattern and locate the Sidebar Articles pattern in the left sidebar
  - [ ] Observe the Sidebar Articles pattern is present in the left sidebar and matches the above criteria

## Visual Regressions

- [x] Run visual regression tests locally to ensure new work does not introduce new bugs.
- [x] Test locally in d8 `develop` to make sure this PR doesn't break the theme.
- [x] Be sure you have created or updated the [scenario in `backstop.json`](ama-style-guide-2/styleguide/backstop.json) and have included a new or updated screenshot in [`bitmaps_reference`](ama-style-guide-2/styleguide/backstop_data/bitmaps_reference).

Note: the new tests added in this PR are expected to fail as there are no live reference images for them yet.

## Relevant Screenshots/GIFs
screencapture-localhost-3000-patterns-03-organisms-sidebar-articles-03-organisms-sidebar-articles-html-2018-03-15-17_08_20

## Remaining Tasks
- [x] Add "Sidebar Articles" to [The Glossary](https://issues.ama-assn.org:8446/confluence/display/DTD/Glossary?src=contextnavpagetreemode).

## Additional Notes
- The "Article Stub as Small Image" variant has lost some styling. I didn't address this here as I felt it was out of scope of this ticket. See [SG1](https://americanmedicalassociation.github.io/AMA-style-guide/?p=molecules-article-preview-small-image) for comparison.

Incorrect appearance
> screen_shot_2018-03-15_at_5_10_29_pm

---

[Guidelines for Contribution](CONTRIBUTING.md)
[SG2 Standards](ama-style-guide-2/docs/standards.md)

### CODE REVIEW

File: styleguide/source/assets/scss/03-organisms/_sidebar-articles.scss [hunk 1]
```diff
@@ -0,0 +1,3 @@
+.ama__sidebar-articles {
+  @include child-top-gutters($gutter / 2);
+}
```
Comment (designerneil) [inline styleguide/source/assets/scss/03-organisms/_sidebar-articles.scss]: just need an empty line here

File: styleguide/source/_patterns/03-organisms/sidebar-articles.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+      class: string / optional
+      paragraph {
+        text:
+          type: string
+      }
+    }
+  ]
+}
+
+~~~
```
Comment (designerneil) [inline styleguide/source/_patterns/03-organisms/sidebar-articles.md]: remove empty line

File: styleguide/source/_patterns/03-organisms/sidebar-articles.json [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+          "text": "A paragraph (from the Greek paragraphos, \"to write beside\" or \"written beside\") is a self-contained unit of a discourse in writing dealing with a particular point or idea. A paragraph consists of one or more sentences. Though not required by the syntax of any language, paragraphs are usually an expected part of formal writing, used to organize longer prose."
+        },
+        "metadata": {
+          "date": "Jul 28, 2017",
+          "readtime": "5 min read"
+        }
+      }
+    ]
+  }
+}
```
Comment (designerneil) [inline styleguide/source/_patterns/03-organisms/sidebar-articles.json]: just need an empty line here

**Comment by labbydev** [comment]
> The "Article Stub as Small Image" variant has lost some styling. I didn't address this here as I felt it was out of scope of this ticket. See SG1 for comparison.
@chelsiejohnston  please be sure to add a bug ticket to address this ^

**Comment by labbydev** [comment]
@chelsiejohnston this visually looks correct when I see it on the Topic page, however I only see a single view mode being used when looking at the Organism itself.

localhost_3000_patterns_03-organisms-sidebar-articles_03-organisms-sidebar-articles html

File: styleguide/source/_patterns/03-organisms/sidebar-articles.json (hunk 3 above)
Comment (labbydev) [inline styleguide/source/_patterns/03-organisms/sidebar-articles.json]: You need an empty line here.
I guess it's intentional that that the organism is all using the same view mode? This is either a problem with the testing instructions or the data here.

File: styleguide/source/_patterns/03-organisms/sidebar-articles.twig [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+  }
+  %}
+
+  {% include "@atoms/heading.twig" %}
+
+  {% for article in sidebarArticles.articles %}
+    {% set articleStub = article %}
+    {% include "@molecules/article-stub.twig" %}
+  {% endfor %}
+</div>
```
Comment (labbydev) [inline styleguide/source/_patterns/03-organisms/sidebar-articles.twig]: Add empty line

File: styleguide/source/assets/scss/03-organisms/_sidebar-articles.scss (hunk 1 above)
Comment (labbydev) [inline styleguide/source/assets/scss/03-organisms/_sidebar-articles.scss]: Add empty line.

**Review by labbydev — CHANGES_REQUESTED** [review]

**Comment by chelsiejohnston** [comment]
Added https://issues.ama-assn.org/browse/EWL-4742 to address the visual regression in the "Article Stub as Small Image" variant.

**Comment by chelsiejohnston** [comment]
@labbydev I believe I've addressed everything mentioned above.

File: styleguide/source/assets/scss/03-organisms/_sidebar-articles.scss (hunk 1 above)
Comment (chelsiejohnston) [reply styleguide/source/assets/scss/03-organisms/_sidebar-articles.scss]: Github doesn't seem to be picking up the empty line changes, but I did add them.

File: styleguide/source/_patterns/03-organisms/sidebar-articles.json (hunk 3 above)
Comment (chelsiejohnston) [reply styleguide/source/_patterns/03-organisms/sidebar-articles.json]: That was not intentional, good catch, it should be fixed now.

Comment (labbydev) [reply styleguide/source/_patterns/03-organisms/sidebar-articles.json]: This has been fixed, thank you!

**Review by labbydev — APPROVED** [review]

**Pull request merged**